isValid() doesn't behave as expected

I'm using `realm-android:0.83.0-SNAPSHOT`.

When I run this simple code:
``` java
Realm realm = Realm.getInstance(this);
realm.beginTransaction();
Dog myDog = realm.createObject(Dog.class);
myDog.setDogName("myDog");
realm.commitTransaction();

realm.beginTransaction();
myDog.removeFromRealm();
realm.commitTransaction();

Timber.e("myDog still valid: " + myDog.isValid());
```

`myDog.isValid()` throws an `IllegalStateException`

```java
     Caused by: java.lang.IllegalStateException: Object is no longer managed by Realm. Has it been deleted?
            at io.realm.internal.InvalidRow.getStubException(InvalidRow.java:180)
            at io.realm.internal.InvalidRow.isAttached(InvalidRow.java:171)
            at io.realm.RealmObject.isValid(RealmObject.java:107)
```

How can I check if an object was removed from realm?
Or does my little example not work because I'm calling this function from the same thread? However, I don't think that `isValid()` should ever throw such an exception.

--- a/changelog.txt
+++ b/changelog.txt
@@ -2,6 +2,7 @@
  * BREAKING CHANGE: Removed deprecated methods and constructors from the Realm class.
  * Fixed a bug which might cause failure when loading the native library.
  * Fixed a bug which might trigger a timeout in Context.finalize().
+ * Fixed a bug which might cause RealmObject.isValid() to throw an exception if the object is deleted.
 
 0.82.1
  * Fixed a bug where using the wrong encryption key first caused the right key to be seen as invalid.

--- a/realm/src/androidTest/java/io/realm/RealmObjectTest.java
+++ b/realm/src/androidTest/java/io/realm/RealmObjectTest.java
@@ -543,4 +543,19 @@ public class RealmObjectTest extends AndroidTestCase {
         } catch (IllegalStateException ignored) {
         }
     }
+
+    public void testIsValid() {
+        testRealm.beginTransaction();
+        Dog dog = testRealm.createObject(Dog.class);
+        dog.setName("Fido");
+        testRealm.commitTransaction();
+
+        assertTrue(dog.isValid());
+
+        testRealm.beginTransaction();
+        dog.removeFromRealm();
+        testRealm.commitTransaction();
+
+        assertFalse(dog.isValid());
+    }
 }

--- a/realm/src/main/java/io/realm/internal/InvalidRow.java
+++ b/realm/src/main/java/io/realm/internal/InvalidRow.java
@@ -20,8 +20,9 @@ package io.realm.internal;
 import java.util.Date;
 
 /**
- * Row wrapper that stubs all access with IllegalStateExceptions. This can be used instead of adding null checks
- * everywhere when the underlying Row accessor in Realm Core is no longer available.
+ * Row wrapper that stubs all access with IllegalStateExceptions except for isAttached. This can be used instead of
+ * adding null checks everywhere when the underlying Row accessor in Realm's underlying storage engine is no longer
+ * available.
  */
 public enum InvalidRow implements Row {
     INSTANCE;
@@ -168,7 +169,7 @@ public enum InvalidRow implements Row {
 
     @Override
     public boolean isAttached() {
-        throw getStubException();
+        return false;
     }
 
     @Override

